gcloud/logging/metrics_service_v2_api.py

```diff
@@ -79,126 +138,85 @@ def __init__(
-            max_attempts=max_attempts,
-            defaults=self.defaults)
-
-    def get_log_metric(
-            self,
-            metric_name='',
-            **kwargs):
-        """Gets a logs-based metric."""
-        get_log_metric_request = logging_metrics_pb2.GetLogMetricRequest(
-            metric_name=metric_name,
-            **kwargs)
-        return self.get_log_metric_callable()(get_log_metric_request)
-
-    def get_log_metric_callable(
-            self,
-            is_retrying=None,
-            max_attempts=None):
-        return api_callable.idempotent_callable(
+            settings=settings)(req,
+                               metadata=self._headers)
+
+    def get_log_metric(self, metric_name='', options=None):
+        """
+        Gets a logs-based metric.
+
+        :type metric_name: string
+        :type options: api_callable.CallOptions
+        """
+        req = logging_metrics_pb2.GetLogMetricRequest(metric_name=metric_name)
+        settings = self._defaults['get_log_metric'].merge(options)
+        return api_callable.ApiCallable(
             self.stub.GetLogMetric,
-            is_retrying=is_retrying,
-            max_attempts=max_attempts,
-            defaults=self.defaults)
-
-    def create_log_metric(
-            self,
-            project_name='',
-            metric=None,
-            **kwargs):
-        """Creates a logs-based metric."""
+            settings=settings)(req,
+                               metadata=self._headers)
+
+    def create_log_metric(self, project_name='', metric=None, options=None):
+        """
+        Creates a logs-based metric.
+
+        :type project_name: string
+        :type metric: logging_metrics_pb2.LogMetric
+        :type options: api_callable.CallOptions
+        """
         if metric is None:
             metric = logging_metrics_pb2.LogMetric()
-        create_log_metric_request = logging_metrics_pb2.CreateLogMetricRequest(
+        req = logging_metrics_pb2.CreateLogMetricRequest(
             project_name=project_name,
-            metric=metric,
-            **kwargs)
-        return self.create_log_metric_callable()(create_log_metric_request)
-
-    def create_log_metric_callable(
-            self,
-            is_retrying=None,
-            max_attempts=None):
-        return api_callable.idempotent_callable(
+            metric=metric)
+        settings = self._defaults['create_log_metric'].merge(options)
+        return api_callable.ApiCallable(
             self.stub.CreateLogMetric,
-            is_retrying=is_retrying,
-            max_attempts=max_attempts,
-            defaults=self.defaults)
-
-    def update_log_metric(
-            self,
-            metric_name='',
-            metric=None,
-            **kwargs):
-        """Creates or updates a logs-based metric."""
+            settings=settings)(req,
+                               metadata=self._headers)
+
+    def update_log_metric(self, metric_name='', metric=None, options=None):
```

Comment:
im a bit confused why you can pass a metric name *and* a metric PB into this. am i just missing something simple ?